python/cudf/dataframe/multiindex.py

```diff
@@ -465,5 +465,29 @@ def from_pandas(cls, multiindex):
                      names=multiindex.names)
         return mi
 
+    @property
+    def is_unique(self):
+        return self.codes._size == self.codes.drop_duplicates()._size
```

Comment:
Instead of using `.codes` here we can use `self._source_data` which will be more efficient.